test/behavior/switch.zig

```diff
@@ -574,6 +574,69 @@ test "switch prongs with cases with identical payload types" {
     try comptime S.doTheTest();
 }
 
+test "switch prong pointer capture alignment" {
+    const U = union(enum) {
+        a: u8 align(8),
+        b: u8 align(4),
+        c: u8,
+    };
+
+    const S = struct {
+        fn doTheTest() !void {
+            const u = U{ .a = 1 };
+            switch (u) {
+                .a => |*a| try expectEqual(*align(8) const u8, @TypeOf(a)),
+                .b, .c => |*p| {
+                    _ = p;
+                    @panic("fail");
+                },
+            }
+
+            switch (u) {
+                .a, .b => |*p| try expectEqual(*align(4) const u8, @TypeOf(p)),
+                .c => |*p| {
+                    _ = p;
+                    @panic("fail");
```

Comment:
These should return an error to make the test fail rather than panic. It's okay, and in fact encouraged, for a behavior test to verify other behaviours it's coincidentally using.